docs/PARSING_DATA.md

```diff
@@ -0,0 +1,216 @@
+broken up into a metric name, tags, fields, and a timestamp. While all of these
+options are not required, they are available to the user and be necessary to
+define to ensure the data is correctly represented.
+
+[line protocol]: https://docs.influxdata.com/influxdb/cloud/reference/syntax/line-protocol/
+
+## Parsers
+
+The first step is to determine which parser to use. Look at the list of
+[parsers][] and find one that will work with the user's data. This is generally
+straightforward as the data-type will only have one parser that is actually
+applicable to the data.
+
+[parsers]: https://github.com/influxdata/telegraf/tree/master/plugins/parsers
+
+### JSON parsers
+
+There is an exception when it comes to JSON data. Instead of a single parser,
+there are three different parsers capable of reading JSON data:
+
+* `json`: This parser is great for flat JSON data. If your JSON is more complex
+  and for example, has other objects or nested arrays, then do not use this and
+  look at the other two options.
+* `json_v2`: The v2 parser was created out of the need to parse JSON objects. It
+  can take on more advanced cases, at the cost of additional configuration.
+* `xpath`: The xpath parser is the most capable of the three options. While the
+  xpath name may imply XML data, it can parse a variety of data types using
+  XPath expressions.
+
+## Tags and fields
+
+The next step is to look at the data and determine how the data needs to be
+split up between tags and fields. Tags are generally strings or values that a
+user will want to search on. While fields are the raw data values, numeric
+types, etc. Generally, data is considered to be a field unless otherwise
+specified as a tag.
+
+## Timestamp
+
+To parser a timestamp, at the very least the users needs to specify which field
+has the timestamp and what the format of the timestamp is. The format can either
+by a predefined Unix timestamp or parsed using a custom format based on Go
+reference time.
+
+For Unix timestamps Telegraf understand the following settings:
+
+| Timestamp             | Timestamp Format |
+|-----------------------|------------------|
+| `1709572232`          | `unix`    |
+| `1709572232123`       | `unix_ms` |
+| `1709572232123456`    | `unix_us` |
+| `1709572232123456789` | `unix_ns` |
+
+To use a custom timestamp, a user must provide the [Go reference time][]
+notation. Here are a few example timestamps and their Go reference time
+equivalent:
+
+| Timestamp                     | Go reference time             |
+|-------------------------------|-------------------------------|
+| `2024-03-04T17:10:32`         | `2006-01-02T15:04:05` |
+| `04 Mar 24 10:10 -0700`       | `02 Jan 06 15:04 -0700` |
+| `2024-03-04T10:10:32Z07:00`   | `2006-01-02T15:04:05Z07:00` |
+| `2024-03-04 17:10:32.123+00`  | `2006-01-02 15:04:05.999+00` |
+| `2024-03-04T10:10:32.123456Z` | `2006-01-02T15:04:05.000000Z` |
+| `2024-03-04T10:10:32.123456Z` | `2006-01-02T15:04:05.999999999Z` |
+
```

Comment:
I think we should also mention timezone handling! Especially if there is no timezone specified in the format (i,e. the time is assumed to be in UTC then if not explicitly given).

Please note: There are also some [named formats](https://github.com/influxdata/telegraf/blob/a6d550e494dcadb1d6d5cb7ff6ca587cbb0604c9/internal/internal.go#L367)...

Reply:
I would maybe leave out the "timezone abbreviations" here, but not sure. I mean parsers are the only place where time-formatting is relevant, so maybe we should put it in here?